### PR TITLE
Support TypeScript 4.7

### LINKED ISSUE #3146: Support TypeScript 4.7
See release annoucement https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/ 

See also #3147

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes #3146

### CODE REVIEW

**Comment by yassin-kammoun-sonarsource** [comment]
The analysis of [mootools-core/Grunt/plugins/karma/syn/lib/syn.js](https://github.com/mootools/mootools-core/blob/master/Grunt/plugins/karma/syn/lib/syn.js) fails because of a [bug](https://github.com/microsoft/TypeScript/issues/49383) in the TypeScript compiler. Something goes wrong while TypeScript tries to infer the type of `__indexOf.call(typeables, fn)` in [there](https://github.com/mootools/mootools-core/blob/187a16bae2d7144ea4b81aa5c0586498f0fe17c7/Grunt/plugins/karma/syn/lib/syn.js#L2995-L2998). The analysis thus fails for this file, and all expected issues are no longer raised.

File: eslint-bridge/package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "mkdirp": "1.0.4",
     "prettier": "2.4.1",
     "ts-jest": "28.0.5",
-    "ts-node": "10.4.0"
+    "ts-node": "10.8.1"
```
Comment (vilchik-elena) [inline eslint-bridge/package.json:43]: You did this just for the sake of updating everything, right? These is no dependency or whatever

File: sonar-javascript-plugin/src/main/resources/static/documentation.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ Discover and update the CSS [properties](/analysis/analysis-parameters/) in: **<
 
 ## Supported Frameworks, Versions and Languages
 * ECMAScript 3, 5, 2015, 2016, 2017, 2018, 2019, and 2020
-* TypeScript 4.6
+* TypeScript 4.7
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/resources/static/documentation.md:30]: no new line 👍  :D

**Review by vilchik-elena — APPROVED** [review]

File: eslint-bridge/package.json (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply eslint-bridge/package.json:43]: Indeed. Now I don't remember why I even upgraded that one.

**Pull request merged**